lib/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -1,72 +1,98 @@
+  id: string;
+  disabled?: boolean;
+  ariaLabel?: string;
+  hideLabel?: boolean;
+  options: T[];
+  defaultValue?: T;
+  selected?: T;
+  onChange?: (value: T) => void;
+}
+
+const DropdownMenuSelect = <T extends DropdownMenuOptionsData = DropdownMenuOptionsData>({
+  id,
+  disabled,
+  ariaLabel,
+  onChange,
+  defaultValue,
+  options,
+}: DropdownMenuSelectProps<T>) => (
+  <select
+    id={id}
+    disabled={disabled}
+    aria-label={ariaLabel}
+    className="min-w-[120px] justify-self-end rounded-lg border border-jod-dark bg-jod-white p-2 hover:bg-purple-100 focus:outline-none focus:ring focus:ring-purple-500 disabled:border-gray-500 disabled:bg-gray-200 disabled:text-gray-500 disabled:hover:bg-gray-200"
+    onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
+      if (onChange) {
+        onChange(options.find((option) => option.value === event.target.value)!);
+      }
+    }}
+    defaultValue={defaultValue?.value}
+  >
+    {options.map((option, index) => (
+      <option key={index} value={option.value}>
+        {option.label}
+      </option>
+    ))}
+  </select>
+);
+
+export interface DropdownMenuOptionsData {
+  value: string;
   label: string;
 }
 
-interface DropdownMenuProps<T extends DropdownMenuOptionsData, U extends string = string> {
+interface DropdownMenuProps<T extends DropdownMenuOptionsData> {
   /** Label for the component */
   label: string;
   /** Hide label. Still available for screenreaders */
   hideLabel?: boolean;
   /** Options for component */
   options: T[];
   /** Default value to be selected initially */
-  defaultValue?: U;
+  defaultValue?: T;
   /** Controlled mode */
-  selected?: U;
+  selected?: T;
   /** Callback on selection change */
-  onChange?: (value: U) => void;
+  onChange?: (value: T) => void;
   /** Component is disabled for user interaction */
   disabled?: boolean;
 }
 
-export const DropdownMenu = <
-  U extends string = string,
-  T extends DropdownMenuOptionsData<string> = DropdownMenuOptionsData<string>,
->({
+export const DropdownMenu = <T extends DropdownMenuOptionsData = DropdownMenuOptionsData>({
   label,
   hideLabel = false,
   options,
   defaultValue,
-  onChange: propOnChange,
+  onChange,
   disabled = false,
-}: DropdownMenuProps<T, U>) => {
-  const labelId = useId();
-  return (
-    <div className="flex flex-row">
-      <label
-        htmlFor={labelId}
-        className={classNames('mr-2 self-center font-bold text-jod-black', {
-          'sr-only': hideLabel,
-        })}
-      >
-        {label}
-      </label>
-      <select
+}: DropdownMenuProps<T>) => {
+  const selectId = useId();
+  return hideLabel ? (
+    <DropdownMenuSelect
+      id={selectId}
+      disabled={disabled}
+      ariaLabel={label}
+      onChange={onChange}
+      defaultValue={defaultValue}
+      options={options}
+    />
+  ) : (
+    <div className="flex items-center gap-2">
+      {!hideLabel && (
```

Comment:
This might be unnecessary to again check `hideLabel` because it should always false in this branch of ternary operation.

Reply:
A good point.